spec/controllers/users/address_autocomplete_controller_spec.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'support/shared/results_shared'
+
+RSpec.describe Users::AddressAutocompleteController, type: :controller do
+  describe 'routes' do
+    it { is_expected.to route(:post, '/api/users/address_autocomplete/complete').to(action: :complete, format: :json) }
+  end
+
+  describe 'POST#complete' do
+    let(:body) do
+      {
+        predictions: [
+          {
+            description: 'Some Place',
+            id: 'Some id',
+            place_id: 'place_id',
+            reference: 'place_id',
+            types: %w[route geocode]
+          },
+          {
+            description: 'Another some place',
+            id: 'Another some id',
+            place_id: 'Another place_id',
+            reference: 'Another place_id',
+            types: %w[route geocode]
+          }
+        ],
+        status: 'OK'
+      }
+    end
+    let(:body_when_missing_input) do
+      {
+        predictions: [],
+        status: 'INVALID_REQUEST'
+      }
+    end
+    let(:params) do
+      {
+        search: {
+          input: input
+        }
+      }
+    end
+    let(:input) { 'some search' }
+    let(:google_response) { OpenStruct.new(body: body.to_json) }
+    let(:subject) { post :complete, params: params, as: :json }
+    before(:each) do
```

Comment:
`before`, by default running before each example, so you can to not pass `:each` arg